### DISCUSSION TITLE
Layer 2 方案的選擇

### DISCUSSION THREAD

**Discussion by yurenju** [lun-dao/LunDAO#11] [General]
等到代幣發送之後，我們第一件事情就是要決定 Layer 2 方案的選擇，想在這邊講一下背景資訊來解釋為什麼我們需要做這個選擇。

# 背景
在目前 Ethereum mainnet 上面執行一筆交易大約是 20 美金，執行 Uniswap 交易則要 80 美金。

![2022 一月 etherscan gas tracker 的截圖](https://user-images.githubusercontent.com/22155/148137160-643f6854-c3fb-40f1-b8b6-542950c5e5c7.png)
*2022 一月 etherscan gas tracker 的截圖*

身為一個寫作獎勵平台，作者獎勵若先預估 1000-5000 台幣時，這樣的交易費就太高了。除此之外我們還會需要對日常的營運工作如引導討論、組織工作事項等等進行獎勵時，有一個符合成本的平台也是非常重要。

# 選擇要素
選擇一個 Layer 2 或是側鏈，針對我們的專案性質有幾點需要列入考量：
- 成本：依照目前作者的獎勵，如果交易成本可以壓在 5 美金以內會比較合適。之後發送時可以一次轉帳代幣給不同的貢獻者，分攤下來或許就可以接受
- 技術：我們的專案目的是鼓勵產生中深度 Ethereum 相關中文文章，此解決方案所採用的技術若符合 Ethereum 未來發展方向會比較合適
- 其他專案活躍度：轉移到該平台後，我們會需要評估治理可能會用到的工具如 gnosis safe, snapshot, coordinape, SourceCred, Disperse.app 是不是可以在該解決方案上正常的運作。

# 已知方案
總之我們會需要討論一下有哪些解決方案，以及大家對於他們的看法來做出最後的決定，先列出我知道的一些解決方案：
- Arbitrum (optimistic rollup)
- optimism (optimistic rollup)
- zkSync (zk rollup)
- xdai (side chain)
- polygon (side chain)

**Comment by gasolin** [comment]
手續費參考 https://l2fees.info/

**Comment by KimiWu123** [comment]
我投 rollup 系列，以手續費與稿費的比例來看，覺得 zk rollup 系列（Loopring or zkSync）比較適合，可以取決於部署, 維運的複雜度

**Reply by yurenju** [reply]
@KimiWu123 你知道 zksync 目前 EVM 的上線時程嗎？另外想要請你發個 PR 在 [contributors.csv][1] 登記你的資訊

[1]: https://github.com/lun-dao/LunDAO/blob/main/contributors.csv

**Reply by KimiWu123** [reply]
目前只有 testnet，印象中之前是說Ｑ１，不過剛剛找了一下，找不到相關時程

**Reply by yurenju** [reply]
如果 zk 還沒好，我覺得可能要選擇生態系比較健全的會比較好，像是 Arbitrum。

好處：
- 流動性高
- 生態系較為健全

壞處：
- 交易費比較高

Optimism 交易費比較少，但是在上面的 dapp 也比較少，只是我們用要到的應該主要有影響的是 gnosis safe, snapshot, Disperse.app 這上面應該都有了。

**Reply by yurenju** [reply]
@KimiWu123 有空發個 PR 登記你的 ethereum account 到 [contributors.csv][1] 歐！

[1]: https://github.com/lun-dao/LunDAO/blob/main/contributors.csv

**Reply by KimiWu123** [reply]
> @KimiWu123 有空發個 PR 登記你的 ethereum account 到 [contributors.csv](https://github.com/lun-dao/LunDAO/blob/main/contributors.csv) 歐！

done! thanks.

**Reply by a2468834** [reply]
zk rollup 要實做 EVM compatible 都還要一段時間 QQQQ
（不然我是滿推 zkp 系列的 project）

倒是有聽說一些非正式的消息：
Loopring 比較偏北京話社群
國外（正常的自由國家）好像 StarkWare 的熱度比較高（？

**Reply by yurenju** [reply]
我覺得 zkSync implement evm 應該不會這麼久 XD
只是應該不會 100% 相容

**Reply by a2468834** [reply]
歐~我的意思就是 100% compatible
p.s. 要有那個膽掛上 compatible 就是要 100% 吧 www

**Comment by yurenju** [comment]
gitcoin 有用 zkSync 耶，不知道他們怎麼 batch transfer
https://www.gushiciku.cn/pl/akHs/zh-tw

![image](https://user-images.githubusercontent.com/22155/148198181-f8e4d42f-04e7-4b7d-9153-fa53299a9ae7.png)

**Reply by gasolin** [reply]
沒有 batch transfer 吧, 每次活動結束後, project owner 還是要自己去 claim 分到的 fund

**Reply by yurenju** [reply]
claim 的意思是已經支援合約了嗎？我對 zkSync 很陌生 🤣

**Reply by gasolin** [reply]
我誤會他們了, 在 gitcoin 上 zkSync 只能拿來收捐款, project owner 還是要在 mainnet 上 claim fund

**Reply by KimiWu123** [reply]
> claim 的意思是已經支援合約了嗎？我對 zkSync 很陌生 🤣

2.0 有支援 EVM 了

**Comment by yurenju** [comment]
統整一下上面的討論：
- zksync 從稿費上來看是最適合我們的 Layer 2，但是目前沒有 EVM，上面也沒有工具可以使用
- Arbitrum 交易費比較貴，但是應用最多，流動性也最高
- Optimism 交易費在上面兩者之間，但是應用跟流動性都比 Arbitrum 少

因為這個關係到之後的其他工作項目如流動性要放哪、社群營運獎勵以及文章獎勵等等事項，所以我想要先做個決定先使用 Arbitrum，因為我們的性質換一個 Layer 2 滿容易的，因為我們只需要工具，但是不會跟其他 app 有太多的交互性。等到 zksync 成熟一點我們再來嘗試看看。

**Reply by a2468834** [reply]
跟架構有關係
事實上在 Arbitrum 部的合約的理論 gas limit 可以超越 L1 block gas limit
Optimism 不行

不過這好像不是需要擔心的一點，反正應該不可能寫出這麼驚人的合約

**Reply by yurenju** [reply]
就實務面來說現在 arbitrum 的 gas fee 滿高的，感覺以目前的狀況解決問題的幅度有限，不過或許之後的更新會解決更多問題
